harness/determined/cli/command.py

```diff
@@ -278,15 +280,32 @@ def config(args: Namespace) -> None:
     print(render.format_object_as_yaml(res_json["config"]))
 
 
-def _set_nested_config(config: Dict[str, Any], key_path: List[str], value: Any) -> Dict[str, Any]:
-    current = config
-    for key in key_path[:-1]:
-        current = current.setdefault(key, {})
-    current[key_path[-1]] = value
-    return config
-
-
-def parse_config_overrides(config: Dict[str, Any], overrides: Iterable[str]) -> None:
+# Convert config overrides in dot notation to json path
+# For example the caller of this func will break down the config override
+#   --config slurm.sbatch_args=[--time=05:00]
+# into key slurm.sbatch_args, value [--time=05:00]
+# And this func will return string in json form:
+# 'slurm': {'sbatch_args': ['--time=05:00']}
+def _dot_to_json(key: Any, value: Any) -> Any:
+    json_output: Any = {}
+    path = key.split(".")
+    target = reduce(lambda d, k: d.setdefault(k, {}), path[:-1], json_output)
+    target[path[-1]] = value
+    return json_output
+
+
+# A recursive function to replace value val in a Dict with a new value rval for override keys
+def _replace_value(
+    data_dict: Dict[str, Any], override_key_path: List[str], val: Any, rval: Any
+) -> None:
+    for key in data_dict.keys():
+        if data_dict[key] == val and key in override_key_path:
```

Comment:
Needs some explanation of what you're trying to accomplish here.

Reply:
added comments